packages/upset/src/components/Header/CollapseAllButton.tsx

```diff
@@ -20,21 +22,31 @@ const collapseAllStyle = css`
 `;
 
 export const CollapseAllButton = () => {
-    const [ collapsedIds, setCollapsedIds ] = useRecoilState(collapsedAtom);
     const firstAggregateBy = useRecoilValue(firstAggregateSelector);
     const dimensions = useRecoilValue(dimensionsSelector);
-
+    const { actions } = useContext(ProvenanceContext);
+    const rows = useRecoilValue(rowsSelector);
     const [ allCollapsed, setAllCollapsed ] = useState(false);
 
     const toggleCollapseAll = () => {
-      const collapsed = !allCollapsed;
-      const ids: {[id: string]: boolean} = {};
-      Object.entries(collapsedIds).forEach((entry) => {
-        ids[entry[0]] = collapsed;
-      })
-  
-      setAllCollapsed(collapsed);
-      setCollapsedIds(ids);
+        const collapsed = !allCollapsed;
+        const ids: string[] = [];
+
+        if (allCollapsed === true) {
+            actions.expandAll();
+            setAllCollapsed(collapsed);
```

Comment:
I think it's a bit hard to follow since you're using collapsed here, and `collapsed = !allCollapsed`. Can we just use the values true and false, and not this variable?